libcylinder/src/key/load.rs

```diff
@@ -242,39 +353,177 @@ mod tests {
         writeln!(temp_file, "{}", private_key.as_hex())
             .expect("Unable to write private key to file");
 
-        let retrieved_private_key =
-            load_user_key(Some(key_name), "TEST_KEY_PATH").expect("Unable retrieve key from file");
+        let retrieved_private_key = load_key(&current_user_key_name(), &current_user_search_path())
+            .expect("Unable retrieve key from file");
 
-        assert_eq!(retrieved_private_key.into_bytes(), private_key.into_bytes(),);
+        assert_eq!(
+            retrieved_private_key.unwrap().into_bytes(),
+            private_key.into_bytes()
+        );
+
+        env::remove_var("CYLINDER_KEY_NAME");
+        env::remove_var("CYLINDER_PATH");
+        assert!(env::var("CYLINDER_KEY_NAME").is_err());
+        assert!(env::var("CYLINDER_PATH").is_err());
+    }
+
+    /// Tests that when the CYLINDER_PATH environment variable is set with nonexisistant paths, and
+    /// no key file exists in the default location `key_load` will return None when passed
+    /// current_user_key_name() and current_user_search_path() as arguments.
+    ///
+    /// 1. Create a private key file in a temporary directory and write a key to the file.
+    /// 2. Set the CYLINDER_KEY_NAME env variable to the name of the private key file.
+    /// 3. Set the CYLINDER_PATH env variable to be two different incorrect file paths.
+    /// 3. Ensure the environment variables were set correctly.
+    /// 4. Call `load_key` with current_user_key_name() and current_user_search_path() as arguments.
+    /// 5. Ensure None is returned.
+    #[test]
+    fn load_key_env_var_bad_paths_none() {
```

Comment:
There is a race condition here because the tests that change environment variables may conflict with each other.

Reply:
Added #[serial(env_var)] to the tests that use the environment variables and added a note about it to the module documentation